File: solution_set1/2255. Count Prefixes of a Given String.py
##ss
class Solution:
    def countPrefixes(self, words: List[str], s: str) -> int:
        
        count = 0
        
        
        def prefix(str1,str2):
            
            if len(str2) < len(str1):
                return False
            
            for x in range(len(str1)):
                if str1[x]!=str2[x]:
                    return False
                
            return True
        
        
        for x in words:
            if prefix(x,s):
                count+=1
                
        return count
        

File: cpp_solutions/2255. Count Prefixes of a Given String.cpp
//ss
class Solution {
public:
    int countPrefixes(vector<string>& words, string s) {
        
        int ans = 0;
        
        for (string comp: words){
            if (isprefix(comp,s)){
                ans+=1;
            }
        }
        return ans;
        
        
    }
    bool isprefix(string s1, string s2){
        //s1 is array string 
        //s2 is the given string
        
        if (s1.length() > s2.length()){
            return false;
        }
        else{
            for(int x=0;x<s1.length();x++){
                if (s1.at(x)!=s2.at(x)){
                    return false;
                }
            }
            return true;
        }
    }
};
